addons/html_editor/static/src/editor/core/collaboration_plugin.js

```diff
@@ -0,0 +1,324 @@
+import { HistoryPlugin } from "./history_plugin";
+
+// 60 seconds
+const HISTORY_SNAPSHOT_INTERVAL = 1000 * 60;
+// 10 seconds
+const HISTORY_SNAPSHOT_BUFFER_TIME = 1000 * 10;
+
+/**
+ * @typedef { Object } CollaborationPluginConfig
+ * @property { string } peerId
+ */
+
+export class CollaborationPlugin extends HistoryPlugin {
+    static name = "history_collaboration";
+    externalStepsBuffer = [];
+    // @todo @phoenix move this in the collaboration plugin ?
+    /** @type { CollaborationPluginConfig['peerId'] } */
+    peerId = null;
+
+    setup() {
+        this.DOMPurify = DOMPurify(this.editable.ownerDocument.defaultView);
```

Comment:
`this.editable.ownerDocument` is also know as `this.document` :stuck_out_tongue: